spring-xd-module/src/main/java/org/springframework/xd/module/ModuleDefinition.java

```diff
@@ -76,21 +73,37 @@ public int compareTo(ModuleDefinition other) {
 	}
 
 	@Override
-	public final boolean equals(Object o) {
-		if (this == o) return true;
-
-		ModuleDefinition that = (ModuleDefinition) o;
-
-		if (!name.equals(that.name)) return false;
-		if (type != that.type) return false;
-
+	public boolean equals(Object obj) {
```

Comment:
Should be careful about this one. IMO the previous implementation better reflects that e.g. can't create/upload module `source:foo` if there is already a `source:foo` (be it composed, or not, etc)